src/partition/repart.c

```diff
@@ -1548,6 +1596,10 @@ static int partition_read_definition(Partition *p, const char *path, const char
                         return log_oom();
         }
 
+        if (p->make_subvolumes && !streq_ptr(p->format, "btrfs"))
+                return log_syntax(NULL, LOG_ERR, path, 1, SYNTHETIC_ERRNO(EINVAL),
+                                  "MakeSubvolumes= can only be used with Format=btrfs, refusing.");
```

Comment:
so in tmpfiles we gracefully degrade if you use "v" and we are not on btrfs. in that case we just create a plain dir instead.

I think we should do the same here, so that people can keep their definitions as generic as possible.

Reply:
Is this really what we want here? As a user, if I specify `MakeSubvolumes=`, I'd want repart to fail if it was unable to create the subvolumes. I think for tmpfiles automatic fallback makes sense because you don't know what filesystem you'll be executing the snippets on, but for repart we know the filesystem type we'll be formatting so we should be explicit here and fail hard if we for some reason can't create the subvolumes on a btrfs filesystem.

Reply:
I'd still gracefully fallback (but log about it with LOG_NOTICE or so), because I think there's value in allowing people to extend vendor supplied default repart files. i.e. there's value in making it easy so that people can use vendor supplied repart files, and make minor modifications by simply adding drop-ins that change the fstype or so while otherwise leaveing the partition definitions of the vendor in place.